microfrontends/another-mfe/public/index.html

```diff
@@ -25,10 +25,42 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>React App</title>
-  </head>
-  <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
+
+    <script src="%REACT_APP_KEYCLOAK_URL%/js/keycloak.js"></script>
     <script>
+      const onLoad = (function() {
+        function createKcDispatcher(payload) {
+          return () => window.dispatchEvent(new CustomEvent('keycloak', { detail: payload }));
+        }
+
+        function initKeycloak() {
+          const keycloak = Keycloak({
+            url: '%REACT_APP_KEYCLOAK_URL%',
+            realm: '%REACT_APP_KEYCLOAK_REALM%',
+            clientId: '%REACT_APP_KEYCLOAK_CLIENT_ID%',
+          });
+
+          keycloak.onReady = createKcDispatcher({ eventType: 'onReady' });
+          keycloak.onAuthSuccess = createKcDispatcher({ eventType: 'onAuthSuccess' });
+          keycloak.onAuthError = createKcDispatcher({ eventType: 'onAuthError' });
+          keycloak.onAuthRefreshSuccess = createKcDispatcher({ eventType: 'onAuthRefreshSuccess' });
+          keycloak.onAuthRefreshError = createKcDispatcher({ eventType: 'onAuthRefreshError' });
+          keycloak.onAuthLogout = createKcDispatcher({ eventType: 'onAuthLogout' });
+          keycloak.onTokenExpired = createKcDispatcher({ eventType: 'onTokenExpired' });
+          const onInit = createKcDispatcher({ eventType: 'onInit' });
+
+          window.entando = {
+            ...(window.entando || {}),
+            keycloak,
+          };
+          window.entando.keycloak
+            .init({ onLoad: 'login-required' })
+            .success(onInit);
+        }
+
+        return initKeycloak;
+      })();
```

Comment:
you will need to add condition to see if `window.entando.keycloak` is now existing. This is because when an MFE is loaded in Portal UI, object `window.entando.keycloak` is already existing. You can try it out on a running app engine `e.g. localhost:8081/entando-de-app/en/my_homepage.page`

Reply:
This is not used in portal-ui, only custom element is

Reply:
`index.html` is just for development/local testing